moshi/moshi/models/tts.py

```diff
@@ -0,0 +1,624 @@
+            else:
+                output = self.token_ids.pad
+        elif token == self.token_ids.word:
+            output = self.token_ids.word
+        elif token == self.token_ids.zero:
+            output = token
+        else:
+            raise RuntimeError(f"Invalid token {token}")
+
+        if self.second_stream_ahead:
+            second = -1
+            if output == self.token_ids.word:
+                # If sampled the `word` special token, we put it on the
+                # second text stream instead of the main one.
+                second = self.token_ids.word
+                if state.queued:
+                    # This allows us to pass the current word tokens faster.
+                    output = state.queued.popleft()
+                else:
+                    output = self.token_ids.pad
+            elif state.lookahead:
+                # Otherwise if we have some lookahead tokens we feed them.
+                second = state.lookahead.popleft()
+            # Then we multiplex the two tokens. We add `+1` to `second` so that
+            # we can encode -1, which would translate to an all 0s embedding.
+            output = (second + 1) * self.token_ids.card + output
+
+        assert output is not None
+        return output, consumed_new_word
+
+
+def script_to_entries(tokenizer: SentencePieceProcessor, token_ids: TokenIds, frame_rate: float,
+                      script: tp.Sequence[str], multi_speaker: bool = True, padding_between: int = 0) -> list[Entry]:
+    """Process a given script into a list of `Entry` that will be consumed by the model.
+
+    This function will perform some replacements such as removing some caracters such as ':', etc.
+    It also supports a single XML tag from SSML, namely `<break time="0.5s">`. This allow the insertion
+    of a pause of roughly the requested duration.
+
+    Args:
+        tokenizer: text tokenizer.
+        token_ids: See `TokenIds`.
+        frame_rate: frame rate of the audio codec.
+        script: list of turns, with each element indicating a change of turn. Starts with the main speaker.
+            Use an empty first turn to start with the other speaker.
+        multi_speaker: whether the model was trained to handle more than one speaker.
+        padding_between: amount of padding to force between words. Will make the model articulate
+            a bit better with values such as 1.
+    """
+    speaker_tokens = [token_ids.main, token_ids.other]
+    last_speaker = None
+    opened_main = False
+    entries = []
+
+    # break is indicated as e.g. <break time="3s"/>
+    event_re = re.compile(r"(?:<break\s+time=\"([0-9]+(?:.[0-9]*)?)s\"\s*/?>)|(?:\s+)")
+
+    def _add_entry(word: str):
+        nonlocal first_content, opened_main, last_speaker
+        assert ' ' not in word
+        assert word
+        tokens = tokenizer.encode(word)  # type: ignore
+        if first_content:
+            speaker = idx % len(speaker_tokens)
```

Comment:
Where is this `idx` defined? Probably this should be added as an `_add_entry` argument and used below?